psamm/tests/test_fastgapfill.py

```diff
@@ -0,0 +1,130 @@
+            f.write('\n'.join([
+                '---',
+                'reactions:',
+                '  - id: rxn_1',
+                '    equation: A[e] <=> B[c]',
+                '  - id: rxn_2',
+                '    equation: A[e] => C[c]',
+                '  - id: rxn_3',
+                '    equation: A[e] => D[e]',
+                '  - id: rxn_4',
+                '    equation: C[c] => D[e]',
+                'compounds:',
+                '  - id: A',
+                '  - id: B',
+                '  - id: C',
+                '  - id: D',
+                'media:',
+                '  - compartment: e',
+                '    compounds:',
+                '      - id: A',
+                '        reaction: rxn_5',
+                '      - id: D',
+                '        reaction: rxn_6',
+                'model:',
+                '  - reactions:',
+                '     - rxn_1',
+                '     - rxn_2',
+            ]))
+        self._model = NativeModel(self._model_dir)
+        self._database = DictDatabase()
+        self._database.set_reaction('rxn_1', parse_reaction('|A| <=>'))
+        self._database.set_reaction('rxn_2', parse_reaction('|A| <=> |B|'))
+        self._database.set_reaction('rxn_3', parse_reaction('|C| <=> |B|'))
+        self._database.set_reaction('rxn_4', parse_reaction('|C| <=>'))
+        self._mm = MetabolicModel.load_model(
+            self._database, self._database.reactions)
+
+
+        try:
+            self._solver = generic.Solver()
+        except generic.RequirementsError:
+            self.skipTest('Unable to find an LP solver for tests')
+
+    def tearDown(self):
+        shutil.rmtree(self._model_dir)
+
+    def test_create_model_complete(self):
+        reactions = set([
+            'rxn_1',
+            'rxn_2',
+            'rxn_3',
+            'rxn_4',
+            'rxn_5',
+            'rxn_6',
+            ('rxntp', Compound('B', 'c')),
+            ('rxntp', Compound('C', 'c')),
+            ('rxnex', Compound('A', 'e')),
+            ('rxnex', Compound('B', 'c')),
+            ('rxnex', Compound('C', 'c')),
+            ('rxnex', Compound('D', 'e'))
+        ])
+
+        w_dict = {
+            'rxn_3': 5.6,
+            'rxn_4': 1.0,
+            ('rxntp', Compound('B', 'c')): 3.0,
+            ('rxntp', Compound('C', 'c')): 3.0,
+            ('rxnex', Compound('A', 'e')): 2.0,
+            ('rxnex', Compound('B', 'c')): 2.0,
+            ('rxnex', Compound('C', 'c')): 2.0,
+            ('rxnex', Compound('D', 'e')): 2.0
+        }
+        penalties = {'rxn_3': 5.6}
+
+        model_complete, weights = fastgapfill.create_model_complete(
+            self._model,
+            db_weight=1.0,
+            ex_weight=2.0,
+            tp_weight=3.0,
+            penalties=penalties)
+        self.assertEqual(set(model_complete.reactions), reactions)
+        self.assertEqual(weights, w_dict)
+
+        def test_fastgapfill():
```

Comment:
This doesn't look right... this test is not run because the indentation makes it a local function defined inside of the other test. I would split these test cases into two classes because the setup is completely different for the two. One requires a NativeModel and the other requires a MetabolicModel.